### DISCUSSION TITLE
Error when `pip install -r requirements-rtd.txt` for contributing to docs

### DISCUSSION THREAD

**Discussion by PaulScemama** [aesara-devs/aesara#1459] [General]
When running the command `pip install -r requirements-rtd.txt`, an issue arises with `use_2to3`. [This](https://stackoverflow.com/questions/69100275/error-while-downloading-the-requirements-using-pip-install-setup-command-use-2) says it is due to `setuptools>=58` breaking support for `use_2to3`. The changelog for `setuptools` notes:

> "[#2086](https://github.com/pypa/setuptools/issues/2086): Removed support for 2to3 during builds. Projects should port to a unified codebase or pin to an older version of Setuptools using [PEP 518](https://peps.python.org/pep-0518/) build-requires."

I resolved the issue by `pip install setuptools==58` and then `pip install -r requirements-rtd.txt`. 

Not entirely sure what the best option is to fix this.

**Comment by brandonwillard** [comment]
In general, it's best to use `conda` to install dependencies in a Conda environment.  We might need to update the documentation if it recommends otherwise.  That said, we need to determine if the `setuptools` issue is due to a bad `environment.yml`.

**Reply by PaulScemama** [reply]
One thing I'll mention is that I've used `environment-arm.yml` since my local machine is an M1 Macbook.
